internal_transfers/actions/tests/e2e/getInternalImbalance.spec.ts

```diff
@@ -16,52 +16,50 @@ async function TxDataFromHash(txHash: string): Promise<MinimalTxData> {
   if (transaction === null) {
     throw new Error(`invalid transaction hash ${txHash} - try again`);
   }
-  const { from, hash, logs } = transaction;
-  const tenderlyLogs = logs.map((log) => {
-    const { address, data, topics } = log;
+  const { from, hash, logs: readonlyLogs } = transaction;
+  const logs = readonlyLogs.map((log) => {
     return {
-      address,
-      data,
+      ...log,
       // had to Map it to make a copy of a readonly field.
-      topics: topics.map((value) => value),
+      topics: log.topics.map((value) => value),
     };
   });
 
   return {
     from,
     hash,
-    logs: tenderlyLogs,
+    logs,
   };
 }
 describe.skip("getInternalImbalance(transaction, simulator)", () => {
-  test.skip("throws when no competition found", async () => {
+  test("throws when no competition found", async () => {
     const txHash =
       "0x08100e7ba81be84ee0bdce43db6640e2f992ec9991a740a689e97d20dea9dafa";
     const transaction = await TxDataFromHash(txHash);
-    await expect(getInternalImbalance(transaction, simulator)).rejects.toThrow(
-      `No competition found for ${txHash}`
-    );
+    await expect(
+      getInternalizedImbalance(transaction, simulator)
+    ).rejects.toThrow(`No competition found for ${txHash}`);
   });
-  test.skip("runs as expected on legit txHash with no imbalance", async () => {
-    const txHash =
-      "0x3b2e9675b6d71a34e9b7f4abb4c9e80922be311076fcbb345d7da9d91a05e048";
-    const transaction = await TxDataFromHash(txHash);
-    const imbalance = await getInternalImbalance(transaction, simulator);
+  test("runs as expected on legit txHash with no imbalance", async () => {
+    const transaction = await TxDataFromHash(
+      "0x3b2e9675b6d71a34e9b7f4abb4c9e80922be311076fcbb345d7da9d91a05e048"
+    );
+    const imbalance = await getInternalizedImbalance(transaction, simulator);
     expect(imbalance).toEqual([]);
   });
-  test.skip("runs as expected on legit txHash - 0xca0bbc", async () => {
-    const txHash =
-      "0xca0bbc3551a4e44c31a9fbd29f872f921548d33400e28debb07ffdc5c2d82370";
-    const transaction = await TxDataFromHash(txHash);
-    const imbalance = await getInternalImbalance(transaction, simulator);
+  test("runs as expected on txHash = 0xca0bbc", async () => {
+    const transaction = await TxDataFromHash(
+      "0xca0bbc3551a4e44c31a9fbd29f872f921548d33400e28debb07ffdc5c2d82370"
+    );
+    const imbalance = await getInternalizedImbalance(transaction, simulator);
     expect(imbalance).toEqual([
       {
         amount: -1608243187495153737n,
         token: "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
       },
       {
         // THIS DISAGREES WITH PYTHON CODE!
-        amount: 230686181n,
+        amount: 218470417n,
```

Comment:
I see that one disagreement has been resolved but this is still a problem. What is the expected value here?

Reply:
I think we should no longer refer to the disagreement (the python code is now operating slightly differently in two different ways).

1. It is running a simulation on `simBlock + 1` while we have changed back to `simBlock`
2. It is computing `|simulatedFullLedger(simBlock+1) - actualLedger(execBlock)|` and we have just changed to compute `|simulatedFullLedger(simBlock) - simulatedReducedLedger(simBlock)| (i.e. the difference of two simulated ledgers on the same block).


I hope this clears up why comparing the two now no longer makes sense.